sudoku/script.js

```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
   }
 });
 btn.addEventListener('click', addRandom);
+
+let checkColumnsArr = [];
+function checkColumns(matrix) {
+  let temp, z;
+  for (let i = 0; i < matrix.length; i++) {
+    checkColumnsArr.splice(0, z);
+    for (let j = 1; j < matrix[i].length; j++) {
+      temp = matrix[0][i];
+      checkColumnsArr.push(matrix[j][i]);
+      z = matrix[j].length - 1;
```

Comment:
Couldn't we combine the checkColumns and checkRows? Mostly they are the same. Depending on a boolean value you can write an `if` for this part and reuse the code

Reply:
sure, will optimize later